blog-posts/optimze-gatsby-urls.md

```diff
@@ -0,0 +1,153 @@
+---
+path: "/blog/how-to-optimize-gatsby-urls/"
+date: "2021-07-15"
+title: "How consistency helps you to optimize Gatsby URLs for users and search engines"
+featuredImage: images/how-to-optimize-gatsby-url-hero.jpeg
+author: "Fabian"
+authorSummary: "Senior Developer at Satellytes"
+attribution:
+  creator: Jonny Clow
+  source: https://unsplash.com/photos/FwMBtl6IQQA
+---
+
+We did a deep dive into how to work with URLs in Gatsby to improve UX and search engine ranking. See what we have 
+learned and how you can improve your Gatsby site too.
+
+<!-- stop excerpt -->
+
+## The problem
+
+If you create a page in Gatsby, the page will be created as `index.html` in a folder that has the same structure as the URL path itself. For example, our position as Senior Frontend engineer has the path `career/325420-senior-frontend-engineer-(mwx)`, which makes Gatsby create an `index.html` in a folder with the same name.
+
+![images/how-to-optimize-gatsby-url-build-folder.png](images/how-to-optimize-gatsby-url-build-folder.png)
+
+Depending on the webserver you are using, the `index.html` file is accessible via different URLs. We are using Netlify which allows 12 different URLs:
+
+- [https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)](https://satellytes.com/career/325420-senior-frontend-engineer-(mwx))
+- [https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/](https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/)
+- [https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/index.html](https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/index.html)
+- all 3 links above, but starting with `https://www.`
+- all 3 links above, but starting with `http://`
+- all 3 links above, but starting with `http://www.`
+
+In general, it's good that all links are supported, as it lowers the possibility of a wrong URL entered by the user. But it also comes with 2 downsides:
+
+1. Degraded performance for the user as only one link returns the HTML file directly. All other links are redirects.
+2. Duplicate content for search engines. A page should only have one URL (or at least one "original" URL)
```

Comment:
```suggestion
In general, it's good that all links are supported, as it lowers the possibility of a wrong URL entered by the user. But it also comes with downsides:

1. We are lost in the freedom of choice and randomly pick any url format because all of them are working.
2. Degraded performance for the user as only one link returns the HTML file directly. All other links are redirects.
3. Duplicate content for search engines. A page should only have one URL (or at least one "original" URL)
```

I added one item to explain the overall problem and how we arrived there. We never decided for a format, netlify was forgiving and everyone picked a random url format becaus it just worked.

Reply:
I applied your point, good one!